backend/apps/progress/views.py

```
@@ -1256,12 +1257,203 @@ def get(self, request):
+                    "breakdown": {
+                        "reading": reading_cnt,
+                        "quizzes": quizzes_cnt,
+                        "code_submissions": code_sub_cnt
+                    }
+                })
 
         return Response(data, status=status.HTTP_200_OK)
 
+
+class HeatmapCSVExportView(APIView):
+    permission_classes = [permissions.IsAuthenticated]
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)